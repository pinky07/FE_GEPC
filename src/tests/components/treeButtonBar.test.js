import React from 'react';
import { mount  } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TreeButtonBar from '../../components/treeButtonBar';

describe('treeButtonBar component', () => {
  const props = {
    save: jest.fn(),
  };
  const wrapper = mount(<TreeButtonBar {...props}/>);

  it ('renders correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it ('the save button should call the save function', () => {
    wrapper.findWhere(n => n.type() === 'button' && n.text() === 'Save').simulate('click');
    expect(props.save).toHaveBeenCalled();
  });

});