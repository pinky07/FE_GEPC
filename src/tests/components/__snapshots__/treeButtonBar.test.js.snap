// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`treeButtonBar component renders correctly 1`] = `
<TreeButtonBar
  save={[Function]}
>
  <span
    className="buttonBar"
  >
    <Button
      color="primary"
      tag="button"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        New
      </button>
    </Button>
    <Button
      color="primary"
      tag="button"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Load
      </button>
    </Button>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </Button>
    <Button
      color="primary"
      tag="button"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Copy
      </button>
    </Button>
    <Button
      color="primary"
      tag="button"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Delete
      </button>
    </Button>
  </span>
</TreeButtonBar>
`;
